x-pack/legacy/plugins/siem/cypress/integration/signal_detection_rules_ml.spec.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { machineLearningRule } from '../objects/rule';
+
+import {
+  ABOUT_FALSE_POSITIVES,
+  ABOUT_MITRE,
+  ABOUT_RISK,
+  ABOUT_SEVERITY,
+  ABOUT_STEP,
+  ABOUT_TAGS,
+  ABOUT_URLS,
+  DEFINITION_TIMELINE,
+  DEFINITION_STEP,
+  RULE_NAME_HEADER,
+  SCHEDULE_LOOPBACK,
+  SCHEDULE_RUNS,
+  SCHEDULE_STEP,
+  ABOUT_RULE_DESCRIPTION,
+  RULE_TYPE,
+  ANOMALY_SCORE,
+  MACHINE_LEARNING_JOB,
+} from '../screens/rule_details';
+import {
+  CUSTOM_RULES_BTN,
+  RISK_SCORE,
+  RULE_NAME,
+  RULE_SWITCH,
+  RULES_ROW,
+  RULES_TABLE,
+  SEVERITY,
+} from '../screens/signal_detection_rules';
+
+import {
+  createAndActivateRule,
+  fillAboutRuleAndContinue,
+  fillDefineMachineLearningRuleAndContinue,
+  selectMachineLearningRuleType,
+} from '../tasks/create_new_rule';
+import {
+  goToManageSignalDetectionRules,
+  waitForSignalsIndexToBeCreated,
+  waitForSignalsPanelToBeLoaded,
+} from '../tasks/detections';
+import {
+  changeToThreeHundredRowsPerPage,
+  filterByCustomRules,
+  goToCreateNewRule,
+  goToRuleDetails,
+  waitForLoadElasticPrebuiltDetectionRulesTableToBeLoaded,
+  waitForRulesToBeLoaded,
+} from '../tasks/signal_detection_rules';
+import { esArchiverLoad, esArchiverUnload } from '../tasks/es_archiver';
+import { loginAndWaitForPageWithoutDateRange } from '../tasks/login';
+
+import { DETECTIONS } from '../urls/navigation';
+
+describe('Signal detection rules, machine learning', () => {
+  before(() => {
+    esArchiverLoad('prebuilt_rules_loaded');
+  });
+
+  after(() => {
+    esArchiverUnload('prebuilt_rules_loaded');
+  });
+
+  it('Creates and activates a new ml rule', () => {
+    loginAndWaitForPageWithoutDateRange(DETECTIONS);
+    waitForSignalsPanelToBeLoaded();
+    waitForSignalsIndexToBeCreated();
+    goToManageSignalDetectionRules();
+    waitForLoadElasticPrebuiltDetectionRulesTableToBeLoaded();
+    goToCreateNewRule();
+    selectMachineLearningRuleType();
+    fillDefineMachineLearningRuleAndContinue(machineLearningRule);
+    fillAboutRuleAndContinue(machineLearningRule);
+    createAndActivateRule();
+
+    cy.get(CUSTOM_RULES_BTN)
+      .invoke('text')
+      .should('eql', 'Custom rules (1)');
+
+    changeToThreeHundredRowsPerPage();
+    waitForRulesToBeLoaded();
+
+    const expectedNumberOfRules = 93;
```

Comment:
Can we derive this from the test data, somehow? This seems brittle as a magic number.

Reply:
in this specific case I prefer to keep the number and don't derive just in case there is a bug loading rules.